File: cpp11/Ice/context/ContextI.cpp
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice/Ice.h>
#include <ContextI.h>

using namespace std;

void
ContextI::call(const Ice::Current& c)
{
    cout << "Type = ";
    const auto p = c.ctx.find("type");
    cout << (p != c.ctx.end() ? p->second : "None") << endl;
}

void
ContextI::shutdown(const Ice::Current& c)
{
    cout << "Shutting down..." << endl;
    c.adapter->getCommunicator()->shutdown();
}

File: swift/Ice/context/Sources/Server/ContextI.swift
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import Ice

class ContextI: Context {
    func call(current: Ice.Current) {
        let type = current.ctx["type"] ?? "None"
        print("Type = \(type)")
    }

    func shutdown(current: Ice.Current) {
        print("Shutting down...")
        current.adapter!.getCommunicator().shutdown()
    }
}
